### PR TITLE
Add timestamp jump detector to Walltime

### LINKED ISSUE #64: Add a check for timestamp jump  to Walltime calculation
Timestamp jump might be caused by the call order of `walltime_.update()`.

In [urg_stamped.cpp](https://github.com/seqsense/urg_stamped/blob/master/src/urg_stamped.cpp)

```
-  cbM()
walltime_.update(scan.timestamp_);
-  cbTM()
walltime_.update(time_device.timestamp_);
- cbII()
walltime_.update(time_device);
```

`walltime_.update` expects its argument to be monotonic increase, however, actually this function is called from different methods in total 3 and its arguments seem not always to be monotonic increase (except 24 bit overflow; 1 << 24 -1 to 0).

For example, the following `update` call order updates `walltime_device_base_`  at 1. and 3.

```
( 0.  time_device_prev_ in walltime.h is 1 << 24 - 1 )
1. walltime_.update(0); 
2. walltime_.update(1 << 24 - 1); 
3. walltime_.update(0);
```

This is because each of the timestamp is gotten by different commands and there is communication delay for each.

Adding check for the timestamp jump is required to avoid it.

### PR DESCRIPTION
**Pull request by ykoga-kyutech**
close #64 

## PR summary
This PR adds a device timestamp jump detector. If the input stamp is largely jumped, the detector regards it as jumped backwards beyond the timestamp boundary (device timestamp is 24 bit).

### CODE REVIEW

**Comment by ykoga-kyutech** [comment]
@at-wat PTAL

File: include/scip2/walltime.h [hunk 1]
```diff
@@ -38,13 +40,34 @@ class Walltime
       initialized_ = true;
     }
 
-    if (time_device < (1 << DEVICE_TIMESTAMP_BITS) / 2 &&
-        (1 << DEVICE_TIMESTAMP_BITS) / 2 < time_device_prev_)
+    if (detectDeviceTimeJump(time_device))
+    {
+      logger::warn() << "Device time jumped." << std::endl;
```
Comment (at-wat) [inline include/scip2/walltime.h:45]: This line seems better to be after L50.

And it would be better to have another time jump check (with more small threshold value) to the places just after calling `Walltime.update()`. (it may be a different PR.)

File: include/scip2/walltime.h [hunk 2]
```diff
@@ -38,13 +40,34 @@ class Walltime
...
+    {
       walltime_device_base_ += 1 << DEVICE_TIMESTAMP_BITS;
+    }
+
     time_device_prev_ = time_device;
 
     return walltime_device_base_ + time_device;
   }
+
+  bool detectDeviceTimeJump(const uint32_t& time_device)
```
Comment (at-wat) [inline include/scip2/walltime.h:64]: I think this is something like `detect...Underflow` rather than `detectDeviceTimeJump`.

File: include/scip2/walltime.h (hunk 1 above)
Comment (ykoga-kyutech) [reply include/scip2/walltime.h]: Yes, `Walltime.update()` just dedicates to convert device time to walltime. 
Time jump check with more small threshold value will be added in another PR.

**Comment by ykoga-kyutech** [comment]
@at-wat Addressed your comments. PTAL
